tests/accelerators/test_hpu.py

```diff
@@ -0,0 +1,358 @@
+
+    os.environ["PL_TORCH_DISTRIBUTED_BACKEND"] = "hccl"
+
+
+class HPUModel(BoringModel):
+    def training_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def test_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def training_epoch_end(self, outputs) -> None:
+        pass
+
+    def validation_epoch_end(self, outputs) -> None:
+        pass
+
+    def test_epoch_end(self, outputs) -> None:
+        pass
+
+
+class HPUClassificationModel(ClassificationModel):
+    def training_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self(x)
+        loss = F.cross_entropy(logits, y)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self(x)
+        acc = self.accuracy(logits, y)
+        return acc
+
+    def test_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self(x)
+        acc = self.accuracy(logits, y)
+        return acc
+
+    def accuracy(self, logits, y):
+        acc = torch.sum(torch.eq(torch.argmax(logits, -1), y).to(torch.float32)) / len(y)
+        return acc
+
+    def validation_epoch_end(self, outputs) -> None:
+        self.log("val_acc", torch.stack(outputs).mean())
+
+    def test_epoch_end(self, outputs) -> None:
+        self.log("test_acc", torch.stack(outputs).mean())
+
+
+@pytest.mark.skipif(_HPU_AVAILABLE, reason="test requires non-HPU machine")
+def test_fail_if_no_hpus(tmpdir):
+    with pytest.raises(MisconfigurationException, match="HPU Accelerator requires HPU devices to run"):
+        Trainer(default_root_dir=tmpdir, hpus=1)
+
+    with pytest.raises(MisconfigurationException, match="HPU Accelerator requires HPU devices to run"):
+        Trainer(default_root_dir=tmpdir, hpus=1, accelerator="hpu")
+
+
+@RunIf(hpu=True)
+def test_accelerator_selected(tmpdir):
+    trainer = Trainer(default_root_dir=tmpdir, hpus=1)
+    assert isinstance(trainer.accelerator, HPUAccelerator)
+    trainer = Trainer(default_root_dir=tmpdir, hpus=1, accelerator="hpu")
+    assert isinstance(trainer.accelerator, HPUAccelerator)
+
+
+@RunIf(hpu=True)
+def test_warning_if_hpus_not_used(tmpdir):
+    with pytest.warns(UserWarning, match="HPU available but not used. Set the `hpus` flag in your trainer"):
+        Trainer(default_root_dir=tmpdir)
+
+
+@RunIf(hpu=True)
+def test_no_warning_plugin(tmpdir):
+    with pytest.warns(None) as record:
+        Trainer(default_root_dir=tmpdir, max_epochs=1, strategy=HPUStrategy(device=torch.device("hpu")))
+    assert len(record) == 0
+
+
+@RunIf(hpu=True)
+def test_all_stages(tmpdir, hpus):
+    model = HPUModel()
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, hpus=hpus)
```

Comment:
```suggestion
    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, accelerator="hpu", devices="auto")
```
Please use the updated accelerator API